tests/models/sam_hq/test_modeling_sam_hq.py

```diff
@@ -0,0 +1,853 @@
+            image_embeddings, intermediate_embeddings = model.get_image_embeddings(pixel_values)
+
+        self.parent.assertEqual(image_embeddings[0].shape, (self.output_channels, 12, 12))
+        self.parent.assertEqual(intermediate_embeddings[0][0].shape, (12, 12, self.hidden_size))
+
+    def create_and_check_get_image_intermediate_embeddings(self, config, pixel_values):
+        model = SamHQModel(config=config)
+        model.to(torch_device)
+        model.eval()
+        with torch.no_grad():
+            image_embeddings, intermediate_embeddings = model.get_image_embeddings(pixel_values)
+
+        self.parent.assertIsInstance(intermediate_embeddings, list)
+        self.parent.assertTrue(len(intermediate_embeddings) > 0)
+        for embedding in intermediate_embeddings:
+            self.parent.assertEqual(embedding.shape, (self.batch_size, 12, 12, self.hidden_size))
+
+    def create_and_check_get_image_hidden_states(self, config, pixel_values):
+        model = SamHQModel(config=config)
+        model.to(torch_device)
+        model.eval()
+        with torch.no_grad():
+            result = model.vision_encoder(
+                pixel_values,
+                output_hidden_states=True,
+                return_dict=True,
+            )
+
+        # after computing the convolutional features
+        expected_hidden_states_shape = (self.batch_size, 12, 12, 36)
+        self.parent.assertEqual(len(result.hidden_states), self.num_hidden_layers + 1)
+        self.parent.assertEqual(result[-1][0].shape, expected_hidden_states_shape)
+
+        with torch.no_grad():
+            result = model.vision_encoder(
+                pixel_values,
+                output_hidden_states=True,
+                return_dict=False,
+            )
+
+        # after computing the convolutional features
+        expected_hidden_states_shape = (self.batch_size, 12, 12, 36)
+        self.parent.assertEqual(len(result[-1]), self.num_hidden_layers + 1)
+        self.parent.assertEqual(result[-1][0].shape, expected_hidden_states_shape)
+
+    def prepare_config_and_inputs_for_common(self):
+        config_and_inputs = self.prepare_config_and_inputs()
+        config, pixel_values = config_and_inputs
+        inputs_dict = {"pixel_values": pixel_values}
+        return config, inputs_dict
+
+
+@require_torch
+class SamHQModelTest(ModelTesterMixin, PipelineTesterMixin, unittest.TestCase):
+    """
+    Here we also overwrite some of the tests of test_modeling_common.py, as SAM-HQ's vision encoder does not use input_ids, inputs_embeds,
+    attention_mask and seq_length.
+    """
+
+    all_model_classes = (SamHQModel,) if is_torch_available() else ()
+    pipeline_model_mapping = (
+        {"feature-extraction": SamHQModel, "mask-generation": SamHQModel} if is_torch_available() else {}
+    )
+    fx_compatible = False
+    test_pruning = False
+    test_resize_embeddings = False
+    test_head_masking = False
+    test_torchscript = False
+
```

Comment:
Here some tests seem broken
```suggestion
test_cpu_offload = False
```